server/src/main/java/org/elasticsearch/common/geo/GeometryTreeReader.java

```diff
@@ -73,7 +73,10 @@ public Extent getExtent() throws IOException {
         if (extent != null) {
             return extent;
         }
-        assert input.readVInt() == 1;
+        int numShapes = input.readVInt();
+        if (numShapes != 1) {
```

Comment:
Maybe it should be `assert numShapes != 1`?

Reply:
I debated, maybe because I was scared. I agree, though. I'll change it to this!